builders/ethereum/precompiles/utility/non-specific.md

```diff
@@ -0,0 +1,91 @@
+
+## Retrieve a Public Key with ECRecoverPublicKey {: verifying-signatures-ecrecoverpublickey }
+
+The primary function of the `ECRecoverPublicKey` precompile is to recover the public key used to create a digital signature from a given message hash and signature. This precompile is similar to [ECRecover](/builders/ethereum/precompiles/utility/eth-mainnet/#verify-signatures-with-ecrecover/){target=\_blank}, with the exception of returning the public key of the account that signed the message rather than the account address. 
+
+In the following sections, you will learn how to use the `ECRecoverPublicKey` precompile.
+
+### Checking Prerequisites {: checking-prerequisites }
+
+--8<-- 'text/_common/install-nodejs.md'
+
+The versions used in this example are v20.15.0 (Node.js) and 10.7.0 (npm). You will also need to install the [Web3](https://web3js.readthedocs.io/en/latest){target=\_blank} package by executing:
+
+```bash
+npm install --save web3
+```
+
+To verify the installed version of Web3, you can use the `ls` command:
+
+```bash
+npm ls web3
+```
+
+This example uses version 4.11.1. You will also use [Remix](/builders/ethereum/dev-env/remix/){target=\_blank}, connecting it to the Moonbase Alpha TestNet via [MetaMask](/tokens/connect/metamask/){target=\_blank}.
+
+--8<-- 'text/_common/endpoint-examples.md'
+
+### Retrieve Transaction Signature Values
+
+To use the `ECRecoverPublicKey` precompile, you must first sign a message to create and retrieve the message hash and transaction signature values (`v`, `r`, `s`) to pass as arguments in the contract call. Please note that the address and private key values used here are hard-coded for simplicity. Always use security best practices when handling private keys. 
+
+Create a new file called `signMessage.js` in your project directory:
+
+```bash
+touch signMessage.js
+```
+
+Open `signMessage.js` in your code editor and add the following script to initialize Web3 with Moonbase Alpha TestNet, sign and hash the message, and return the signature values:
+
+```js title="signMessage.js"
+--8<-- 'code/builders/ethereum/precompiles/utility/nonspecific/signMessage.js'
+```
+
+Return to your terminal command line to run the script with this command:
+
+```bash
+node signMessage.js
+```
+
+This code will return the following object in the terminal:
+
+--8<-- 'code/builders/ethereum/precompiles/utility/nonspecific/terminal/signature.md'
+
+Save these values as you will need them in the next section.
+
+### Test ECRecoverPublicKey Contract
+
+You can now visit [Remix](/builders/ethereum/dev-env/remix/){target=\_blank} to test the precompiled contract. Note that you could also use the Web3.js library, but in this case, you can go to Remix to ensure it is using the precompiled contract on the blockchain. The Solidity code you can use to retrieve the public key is the following:
```

Comment:
I was kind of expecting the Remix link to take me to Remix itself but either is okay!

Reply:
Or maybe add Remix to the next link?